src/battle_util.c

```diff
@@ -7835,25 +7852,32 @@ bool32 DoBattlersShareType(u32 battler1, u32 battler2)
 bool32 CanBattlerGetOrLoseItem(u8 battlerId, u16 itemId)
 {
     u16 species = gBattleMons[battlerId].species;
-
-    if (IS_ITEM_MAIL(itemId))
-        return FALSE;
-    else if (itemId == ITEM_ENIGMA_BERRY)
+    u16 holdEffect = ItemId_GetHoldEffect(itemId);
+    
+    // Mail can be stolen now
+    if (itemId == ITEM_ENIGMA_BERRY)
         return FALSE;
-    else if (species == SPECIES_KYOGRE && itemId == ITEM_BLUE_ORB)
+    else if (GET_BASE_SPECIES_ID(species) == SPECIES_KYOGRE && (itemId == ITEM_BLUE_ORB || holdEffect == HOLD_EFFECT_PRIMAL_ORB))
         return FALSE;
-    else if (species == SPECIES_GROUDON && itemId == ITEM_RED_ORB)
+    else if (GET_BASE_SPECIES_ID(species) == SPECIES_GROUDON && (itemId == ITEM_BLUE_ORB || holdEffect == HOLD_EFFECT_PRIMAL_ORB))
```

Comment:
Groudon's check should use ITEM_RED_ORB.

Also, isn't it a bad idea to do an OR check with HOLD_EFFECT_PRIMAL_ORB because then it would make it so that Kyogre can't steal a Red Orb or lose it and vice versa for Groudon?

Reply:
fixed the Red Orb issue. Primal orbs use are red/blue orb so I am assuming for now that just checking these is sufficient